tests/common/plugins/sanity_check/checks.py

```diff
@@ -96,23 +96,39 @@ def check_interfaces(dut):
     logger.info("Done checking interfaces status.")
     return check_result
 
+
 def check_bgp_status(dut):
+
+    def _check_bgp_status_helper():
+        neighbors.clear()
+        neighbors.update(dut.get_bgp_neighbors())
+        if neighbors:
+            down_neighbors = [k for k, v in neighbors.items()
+                              if v['state'] != 'established']
+            if down_neighbors:
+                check_result['down_neighbors'] = down_neighbors
+                check_result['failed'] = True
+            else:
+                check_result['failed'] = False
+                check_result.pop('down_neighbors', None)
+        else:
+            check_result['failed'] = True
+        return not check_result['failed']
+
     logger.info("Checking bgp status...")
     check_result = {"failed": False, "check_item": "bgp"}
+    networking_uptime = dut.get_networking_uptime().seconds
+    timeout = max(SYSTEM_STABILIZE_MAX_TIME - networking_uptime, 0)
+    interval = 20
+    neighbors = {}
 
-    neis = dut.get_bgp_neighbors()
-    if not neis:
-        logger.info("BGP neighbors: None")
-        check_result["failed"] = True
+    wait_until(timeout, interval, _check_bgp_status_helper)
+    if neighbors:
```

Comment:
neighbors will be empty according to the current logic

Reply:
Updated, set `timeout` to at least one sec to try at least once.